editor/components/block-list/block.js

```diff
@@ -348,6 +354,96 @@ export class BlockListBlock extends Component {
 		this.setState( { error } );
 	}
 
+	onDragStart( event ) {
+		const dragInset = document.getElementById( `block-drag-inset-${ this.props.uid }` );
+		const block = document.getElementById( `block-${ this.props.uid }` );
+
+		// Closure to remove the cloned node from the DOM (fired within timeout below)
+		const removeBlockClone = ( _blockList, _cloneWrapper ) => {
+			return () => {
+				_blockList.removeChild( _cloneWrapper );
+			};
+		};
+
+		// Closure to hide the visible block and show inset in its place (fired within timeout below)
+		const showInset = () => {
+			return () => {
+				this.setState( state => Object.assign( {}, state, { dragging: true } ) );
+				document.body.classList.add( 'dragging' );
+			};
+		};
+
+		event.dataTransfer.setData(
+			'text',
+			JSON.stringify( {
+				uid: this.props.uid,
+				fromIndex: this.props.order,
+				type: BLOCK_REORDER,
+			} )
+		);
+
+		if ( 'function' === typeof event.dataTransfer.setDragImage ) {
```

Comment:
In what cases this won't be true?

Reply:
This block makes `onDragStart` quite long and hard to follow for me – its purpose is not obvious until I have reached the middle of it (moving the clone around).

Reply:
> In what cases this won't be true?

Oh man, don't ask. IE11!

> This block makes onDragStart quite long and hard to follow for me – its purpose is not obvious until I have reached the middle of it (moving the clone around).

Totally agree this will be refactored in the end.